### PR TITLE
LGPE & Isle of Armor move data.

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Added missing Gen 7 & 8 move data.

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

**Comment by ExpoSeed** [comment]
Holding off on merging this until we figure out our project boards

File: data/battle_anim_scripts.s [hunk 1]
```diff
@@ -692,6 +692,19 @@ gBattleAnims_Moves::
...
+    .4byte Move_PIKA_POW
+    .4byte Move_BOUNCY_BUBBLE
+    .4byte Move_BUZZY_BUZZ
+    .4byte Move_SIZZLY_SLIDE
+    .4byte Move_GLITZY_GLOW
+    .4byte Move_BADDY_BAD
+    .4byte Move_SAPPY_SEED
+    .4byte Move_FREEZY_FROST
+    .4byte Move_SPARKLY_SWIRL
+    .4byte Move_VEEVEE_VOLLEY
```
Comment (LOuroboros) [inline data/battle_anim_scripts.s:707]: Small nitpick, but you're using the wrong indentation style for these new moves. Should be tabs, not spaces.

File: data/battle_anim_scripts.s [hunk 2]
```diff
@@ -729,7 +742,29 @@ gBattleAnims_Moves::
...
+    .4byte Move_POLTERGEIST
+    .4byte Move_CORROSIVE_GAS
+    .4byte Move_COACHING
+    .4byte Move_FLIP_TURN
+    .4byte Move_TRIPLE_AXEL
+    .4byte Move_DUAL_WINGBEAT
+    .4byte Move_SCORCHING_SANDS
+    .4byte Move_JUNGLE_HEALING
+    .4byte Move_WICKED_BLOW
+    .4byte Move_SURGING_STRIKES
```
Comment (LOuroboros) [inline data/battle_anim_scripts.s:766]: Just like above, the wrong indentation style is being used here. Should be tabs, not spaces.

File: src/data/battle_moves.h [hunk 3]
```diff
@@ -9486,6 +9486,188 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
+        .priority = 0,
+        .flags = FLAG_MAKES_CONTACT | FLAG_PROTECT_AFFECTED,
+        .split = SPLIT_PHYSICAL,
+    },
+
+    [MOVE_PIKA_POW] =
+    {
+        .effect = EFFECT_RETURN,
+        .power = 1,
+        .type = TYPE_FLYING,
```
Comment (Toby313) [inline src/data/battle_moves.h:9535]: Pika Papow should be Electric type and is spelled incorrectly.

**Comment by AsparagusEduardo** [comment]
Fixes applied.

**Review by ExpoSeed — DISMISSED** [review]
Issues will need to be made for the effects marked `TODO`

File: src/data/text/move_descriptions.h [hunk 4]
```diff
@@ -2766,10 +2818,97 @@ static const u8 sSTEEL_BEAMDescription[] = _(
...
+    "Shoots scales 2 to 5 times.\n"
+    "Ups Speed, lowers defense.");
+
+static const u8 sMETEOR_BEAMDescription[] = _(
+    "A 2-turn move that raises\n"
+    "Sp. Attack before attacking.");
+
+static const u8 sSHELL_SIDE_ARMDescription[] = _(
+	"Its category depends on\n"
+	"which one will damage more.");
```
Comment (MandL27) [inline src/data/text/move_descriptions.h:2839]: Inconsistent indentation (also on a few other moves)

File: src/data/text/move_descriptions.h [hunk 5]
```diff
@@ -2618,9 +2618,61 @@ static const u8 sPLASMA_FISTSDescription[] = _(
 	"Normal moves become Electric.");
 
 static const u8 sPHOTON_GEYSERDescription[] = _(
-	"The foe's highest offensive\n"
+	"The user's highest offensive\n"
```
Comment (MandL27) [inline src/data/text/move_descriptions.h:2621]: Feel like either this or the next line is missing "stat"; maybe "The user's highest stat determines its category" if space is a concern

**Review by MandL27 — COMMENTED** [review]
Overall looks good. Part of me wants to PR my battle menu rework so we could fit longer move names, but having such a UI-breaking change in a rombase sounds...not wise.

File: src/data/text/move_descriptions.h [hunk 6]
```diff
@@ -2766,10 +2818,97 @@ static const u8 sSTEEL_BEAMDescription[] = _(
...
+static const u8 sJUNGLE_HEALINGDescription[] = _(
+    "Becomes one with the jungle\n"
+    "to heal the team's health.");
+
+static const u8 sWICKED_BLOWDescription[] = _(
+    "By mastering the Dark style,\n"
+    "always does a critical hit.");
+
+static const u8 sSURGING_STRIKESDescription[] = _(
+    "The Water style allows to\n"
```
Comment (ExpoSeed) [inline src/data/text/move_descriptions.h:2906]: "allows to do 3 critical hits" is awkward phrasing imo. I'd probably do "strikes with 3 critical hits" or "lands with 3 critical hits"

**Review by ExpoSeed — CHANGES_REQUESTED** [review]
i forgot to hit request changes

**Review by LOuroboros — DISMISSED** [review]
Lgtm.

**Review by ExpoSeed — APPROVED** [review]

File: src/data/text/move_names.h [hunk 7]
```diff
@@ -713,4 +726,26 @@ const u8 gMoveNames[MOVES_COUNT][MOVE_NAME_LENGTH + 1] =
...
     [MOVE_ETERNABEAM] = _("Eternabeam"),
     [MOVE_STEEL_BEAM] = _("Steel Beam"),
+    [MOVE_EXPANDING_FORCE] = _("Expand Force"),
+    [MOVE_STEEL_ROLLER] = _("Steel Roller"),
+    [MOVE_SCALE_SHOT] = _("Scale Shot"),
+    [MOVE_METEOR_BEAM] = _("Meteor Beam"),
+    [MOVE_SHELL_SIDE_ARM] = _("Shell S. Arm"),
+    [MOVE_MISTY_EXPLOSION] = _("Misty Explos"),
+    [MOVE_GRASSY_GLIDE] = _("Grassy Glide"),
+    [MOVE_RISING_VOLTAGE] = _("Rise Voltage"),
```
Comment (ultima-soul) [inline src/data/text/move_names.h:736]: Maybe `Rising Volts`?

File: src/data/battle_moves.h [hunk 8]
```diff
@@ -10316,9 +10316,243 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
+        .type = TYPE_WATER,
+        .accuracy = 100,
+        .pp = 15,
+        .secondaryEffectChance = 30,
+        .target = MOVE_TARGET_BOTH,
+        .priority = 0,
+        .split = SPLIT_SPECIAL,
+    },
+
+    [MOVE_FLOATY_FALL] =
```
Comment (ultima-soul) [inline src/data/battle_moves.h:10355]: A check should be added in `IsGravityPreventingMove` since it is affected by Gravity.

**Review by ultima-soul — CHANGES_REQUESTED** [review]
There are some duplicates of issues mentioned above for newline and empty line stuff so make sure to cover those. 

Feel free to give opinions on the move descriptions I suggested. They were done to be a bit more closer to official descriptions alongside fixing some errors.

File: src/data/battle_moves.h [hunk 9]
```diff
@@ -10820,4 +11054,312 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
+        .secondaryEffectChance = 0,
+        .target = MOVE_TARGET_SELECTED,
+        .priority = 0,
+        .flags = FLAG_MAKES_CONTACT | FLAG_PROTECT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED | FLAG_KINGSROCK_AFFECTED,
+        .split = SPLIT_PHYSICAL,
+    },
+
+    [MOVE_TRIPLE_AXEL] =
+    {
+        .effect = EFFECT_PLACEHOLDER,   //TODO
```
[the remark this answers is not in the document]
Comment (ExpoSeed) [reply src/data/battle_moves.h:11284]: bulba says triple axel increases by 20 each time as opposed to triple kick's 10

**Comment by AsparagusEduardo** [comment]
Review changes ready

**Review by ghoulslash — DISMISSED** [review]

File: src/data/battle_moves.h [hunk 10]
```diff
@@ -10316,15 +10316,271 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
+    [MOVE_ZIPPY_ZAP] =
+    {
+        #if B_UPDATED_MOVE_DATA >= GEN_8
+            .power = 80,
+            .pp = 10,
+            .flags = FLAG_MAKES_CONTACT | FLAG_PROTECT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED,
+        #else
+            .power = 50,
+            .pp = 15,
+            .flags = FLAG_MAKES_CONTACT | FLAG_PROTECT_AFFECTED,
```
Comment (ultima-soul) [inline src/data/battle_moves.h:10328]:
```suggestion
            .flags = FLAG_MAKES_CONTACT | FLAG_PROTECT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED | FLAG_KINGSROCK_AFFECTED,
        #else
            .power = 50,
            .pp = 15,
            .flags = FLAG_MAKES_CONTACT | FLAG_PROTECT_AFFECTED | FLAG_KINGSROCK_AFFECTED,
```

File: src/data/battle_moves.h [hunk 11]
```diff
@@ -10316,15 +10316,271 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
+        .priority = 2,
+        .split = SPLIT_PHYSICAL,
+    },
+
+    [MOVE_SPLISHY_SPLASH] =
+    {
+        #if B_UPDATED_MOVE_DATA >= GEN_8
+            .flags = FLAG_PROTECT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED,
+        #else
+            .flags = FLAG_PROTECT_AFFECTED,
```
Comment (ultima-soul) [inline src/data/battle_moves.h:10344]:
```suggestion
            .flags = FLAG_PROTECT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED | FLAG_KINGSROCK_AFFECTED,
        #else
            .flags = FLAG_PROTECT_AFFECTED | FLAG_KINGSROCK_AFFECTED,
```

File: src/data/battle_moves.h [hunk 12]
```diff
@@ -10316,15 +10316,271 @@ const struct BattleMove gBattleMoves[MOVES_COUNT] =
...
+        #if B_UPDATED_MOVE_DATA >= GEN_8
+            .power = 100,
+            .accuracy = 90,
+            .pp = 10,
+            .flags = FLAG_PROTECT_AFFECTED | FLAG_MAGICCOAT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED,
+        #else
+            .power = 90,
+            .accuracy = 100,
+            .pp = 15,
+            .flags = FLAG_PROTECT_AFFECTED | FLAG_MAGICCOAT_AFFECTED,
```
Comment (ultima-soul) [inline src/data/battle_moves.h:10505]:
```suggestion
            .flags = FLAG_PROTECT_AFFECTED | FLAG_MAGICCOAT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED | FLAG_KINGSROCK_AFFECTED,
        #else
            .power = 90,
            .accuracy = 100,
            .pp = 15,
            .flags = FLAG_PROTECT_AFFECTED | FLAG_MAGICCOAT_AFFECTED | FLAG_KINGSROCK_AFFECTED,
```

**Review by ultima-soul — CHANGES_REQUESTED** [review]
Unnecessary newlines and non-empty ones seemed to still be present in battle_anim_scripts.s

**Comment by AsparagusEduardo** [comment]
Re-re-ready for re-re-review

**Review by ultima-soul — APPROVED** [review]

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**